src/coarse/coarse.c

```diff
@@ -1030,6 +1030,11 @@ umf_result_t coarse_alloc(coarse_t *coarse, size_t size, size_t alignment,
         alignment = ALIGN_UP(alignment, coarse->page_size);
     }
 
+    if (size + alignment < size) {
+        LOG_ERR("size + alignment overflow");
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
```

Comment:
[Just a reminder] It should not be needed after having merged https://github.com/oneapi-src/unified-memory-framework/pull/981